pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -53,11 +53,15 @@ func (p *PodConverter) PodToDataplane(
 
 func (p *PodConverter) PodToIngress(dataplane *mesh_k8s.Dataplane, pod *kube_core.Pod, services []*kube_core.Service) error {
 	dataplane.Mesh = MeshFor(pod)
```

Comment:
Ok, what do you suggest?

Reply:
Apparently here: 
```go
operationResult, err := kube_controllerutil.CreateOrUpdate(ctx, r.Client, ingress, func() error {
	if err := r.PodConverter.PodToIngress(ingress, pod, services); err != nil {
		return errors.Wrap(err, "unable to translate a Pod into a Ingress")
	}
	if err := kube_controllerutil.SetControllerReference(pod, ingress, r.Scheme); err != nil {
		return errors.Wrap(err, "unable to set Ingress's controller reference to Pod")
	}
	return nil
})
```
we need to copy `ingress` before passing it to `PodToIngress`.

But I don't really like this approach. Another option could be to get rid of `kube_controllerutil.CreateOrUpdate`, it has a pretty simple and straightforward implementation. We can do the same but with DeepCopy. Actually, if you look at the implementation it already has DeepCopy, but they use it to avoid additional Update.